### PR TITLE
feat(Gallery): add min/max widths at breakpoints

### LINKED ISSUE #5055: Gallery layout supports customizable min/max-width
PRs:
* https://github.com/patternfly/patternfly/pull/3549/
* https://github.com/patternfly/patternfly/pull/3703/

You can override the column width by setting `--pf-l-gallery--GridTemplateColumns--min{-on-[breakpoint]}` and `--pf-l-gallery--GridTemplateColumns--max{-on-[breakpoint]}`

Examples are on https://pf4.patternfly.org/layouts/gallery

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #5055

- Adds `minWidths` and `maxWidths` properties that specify their respective width modifier at different breakpoints. The `default` will set a min or max width in general (not at any breakpoint).

### CODE REVIEW

**Review by nicolethoen — DISMISSED** [review]

**Review by mattnolting — DISMISSED** [review]
LPTM, great work @kmcfaul!

File: packages/react-integration/demo-app-ts/src/components/demos/GalleryDemo/GalleryDemo.tsx [hunk 1]
```diff
@@ -5,7 +5,18 @@ export class GalleryDemo extends React.Component {
   static displayName = 'GalleryDemo';
   render() {
     return (
-      <Gallery hasGutter>
+      <Gallery
```
Comment (tlabaj) [inline packages/react-integration/demo-app-ts/src/components/demos/GalleryDemo/GalleryDemo.tsx:8]: I would add demo for each breakpoint.  We have instances in the past where certain breakpoints styles were not being applied as expected.

**Review by tlabaj — CHANGES_REQUESTED** [review]

File: packages/react-integration/demo-app-ts/src/components/demos/GalleryDemo/GalleryDemo.tsx (hunk 1 above)
Comment (kmcfaul) [reply packages/react-integration/demo-app-ts/src/components/demos/GalleryDemo/GalleryDemo.tsx:8]: I'm confused, by each breakpoint do you mean one demo for `minWidths` and one demo for `maxWidths` or do you mean one demo per size (sm/md/lg/etc) per property? The current test is testing the whole style object to ensure the classes are applied properly. Is there a different method of testing I should use in the cypress tests instead?

Comment (tlabaj) [reply packages/react-integration/demo-app-ts/src/components/demos/GalleryDemo/GalleryDemo.tsx:8]: @kmcfaul I would just add the missing breakpoints to what you have (xl and 2xl etc) for both the max and min widths to make sure they are applied correctly.

Comment (kmcfaul) [reply packages/react-integration/demo-app-ts/src/components/demos/GalleryDemo/GalleryDemo.tsx:8]: Oh I understand, gotcha!

**Comment by kmcfaul** [comment]
Updated to check all breakpoints in integration

**Review by tlabaj — APPROVED** [review]
Looks great!

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**